site_scons/site_tools/compiler_setup.py

```diff
@@ -55,6 +55,9 @@ def _base_setup(env):
     env.Append(CCFLAGS=['-g', '-Wextra', '-Wshadow', '-Wall', '-fpic'])
 
     env.AppendIfSupported(CCFLAGS=DESIRED_FLAGS)
+    env.AppendIfSupported(LINKFLAGS=["-fstack-protector-strong"])
+    if '-fstack-protector-strong' in env["CCFLAGS"]:
+        env.AppendENVPath("CGO_LDFLAGS", "-fstack-protector-strong", sep=" ")
```

Comment:
This part of the PR could go into master I assume?

Reply:
This is the one part I'm not sure about. It only seems to happen on Frontera, and only after I started using `SCONS_ENV=full`, so I wonder if the core problem is some mangling of the env (gcc?)

Reply:
`AppendIfSupported` is one of ours and only checks `CFLAGS` I think.  It also makes some assumptions about c vs c++ which are incorrect.

Reply:
I'd like to understand why this line is need on Frontera before pushing upstream. FWIW, some time ago some of your builds changes magically fixed some other linkage issue I was running into. But I don't remember which change(s) or the specific issue. So I've been holding out on this one

Reply:
Closing this PR is favor of #13201 until I can figure out the proper fix for this line